### ISSUE TITLE
Upgrading from node v 4.2.6 to 6+

### ISSUE THREAD

**Issue by dotnetspec** [nodejs/help#607]
* **Node.js Version**: 4.2.6
* **OS**: Linux Mint 18
* **Scope (install, code, runtime, meta, other?)**: install/update
* **Module (and version) (if relevant)**: N/A

I have same problem as #339 (I can't upgrade from version 4+ to 6+), but restart didn't resolve

![node version](https://cloud.githubusercontent.com/assets/27340988/25511685/e540b9d4-2bfa-11e7-9264-734c7e3699ca.png)

make
make install
were run without error in the downloaded node-v6 folder

**Comment by gibfahn** [comment]
How did you download the `node-v6` folder?

By default `make install` installs to `/usr/local/bin/node` and `/usr/local/bin/npm`, do those binaries exist? I'm guessing they don't, as `/usr/local/bin` is before `/usr/bin` in your PATH, but it's worth checking. Did you specify a different prefix?

**Comment by gibfahn** [comment]
If you go to the `node-v6` source folder, as you've already run make you should see a `node` symlink in the root directory, should look like:

```
node -> out/Release/node
```

Can you check that that is Node 6 (`./node -v`)?

**Comment by dotnetspec** [comment]
How did you download the node-v6 folder?
 - https://nodejs.org/en/download/

By default make install installs to /usr/local/bin/node and /usr/local/bin/npm, do those binaries exist? 
- no
I'm guessing they don't, as /usr/local/bin is before /usr/bin in your PATH, but it's worth checking. 

Did you specify a different prefix in one of the three steps (./configure; make -j4; make install)?
- I did not ./configure initially. I did not add -j4 flag

If you go to the node-v6 source folder, as you've already run make you should see a node symlink in the root directory, should look like:
node -> out/Release/node
Can you check that that is Node 6 (cd /path/to/node-v6 && ./node -v)?

- I decided to download the source files (instead of the original binary). On running 'make -j4' it appears to go into a loop (I waited at least 25 min):

![make -j4](https://cloud.githubusercontent.com/assets/27340988/25523254/99798ace-2c38-11e7-9317-d97fe7f658da.png)

and there is now still no symlink in node-v6 source folder:

![node 6 root](https://cloud.githubusercontent.com/assets/27340988/25523716/5a377d7e-2c3a-11e7-89f1-d23ad5ecad8c.png)

Thanks for your patience!

**Comment by gibfahn** [comment]
Here are my standard installation suggestions for Node on Linux, if you're happy to get Node from a tarball or a repo, use them. If you want to build from source see the next comment:

If you want to use a package manager, see [this guide](https://nodejs.org/en/download/package-manager/). For older Linux distros you might need to see [these instructions](https://github.com/nodesource/distributions/blob/master/OLDER_DISTROS.md).

If you want to install node manually, there is no "installation" on Linux per se, you just need to extract the tarball, and make sure the `node` and `npm` binaries in the `./bin` subdirectory are somewhere in your path.

I find [nvm](https://github.com/creationix/nvm#install-script) is useful for managing node installations (it will download, extract, and add to path for you). If you want to do things manually, you can do something like:

```bash
wget https://nodejs.org/dist/v6.9.2/node-v6.9.2-linux-x64.tar.gz
tar -xf node-v6.9.2-linux-x64.tar.gz --directory /usr/local --strip-components 1
```

This will put the `./bin` directory in `/usr/local/bin`, which should be in your `$PATH` already. You can verify that it worked with:

```bash
node --version
npm --version
```

**Comment by gibfahn** [comment]
>I decided to download the source files (instead of the original binary). On running 'make -j4' it appears to go into a loop (I waited at least 25 min):

Depending on how fast your computer is, it can take well over half an hour to build node. If you want to build from source then you'll have to wait till it finishes. The last commands are:

```bash
make[2]: Leaving directory '/home/iojs/build/workspace/node-test-commit-linux/nodes/ubuntu1604-64/out'
if [ ! -r node -o ! -L node ]; then ln -fs out/Release/node node; fi
```

If you want to see what the build process looks like, you can look at our ci, here's [the last run on Linux](https://ci.nodejs.org/job/node-test-commit-linux/nodes=ubuntu1604-64/lastBuild/consoleText). Everything up to the above two lines is the build.

I recommend using the prebuilt tarballs unless you really want to build from source though, that's why we have so many builds!

**Comment by dotnetspec** [comment]
I was having difficulties with the various suggestions. However, I did a system update and a new node.js upgrade for v.6 suddenly appeared in the Update Manager. I simply installed through that and the system recognized v.6. I had done an earlier install of v.4.2.6 through UM. It therefore appears the system is giving some kind of preference to UM installs since the later ones I was doing myself based on your advice weren't working.

Thanks again for your help. e.g. stating
if [ ! -r node -o ! -L node ]; then ln -fs out/Release/node node; fi
as the last command let me move ahead where I may have been unsure.

There is now a new issue with dapp init which I will raise in another ticket
